src/legacy/core_plugins/kibana/public/management/sections/index_patterns/create_index_pattern_wizard/components/step_time_field/step_time_field.js

```diff
@@ -109,10 +118,45 @@ export class StepTimeFieldComponent extends Component {
     }));
   }
 
-  createIndexPattern = () => {
+  createIndexPattern = async () => {
+    const { createIndexPattern } = this.props;
     const { selectedTimeField, indexPatternId } = this.state;
     this.setState({ isCreating: true });
-    this.props.createIndexPattern(selectedTimeField, indexPatternId);
+    try {
+      await createIndexPattern(selectedTimeField, indexPatternId);
+    } catch (error) {
+      if (!this.mounted) return;
+      this.setState({
+        error: error instanceof Error ? error.message : String(error)
+      });
+    } finally {
+      if (this.mounted) {
+        this.setState({ isCreating: false });
+      }
+    }
+  }
+
+  formatErrorMessage(message) {
+    // `createIndexPattern` throws "Conflict" when index pattern ID already exists.
+    return message === 'Conflict'
+      ? (
+        <FormattedMessage
+          id="kbn.management.createIndexPattern.stepTime.patterAlreadyExists"
+          defaultMessage="Custom index pattern ID already exists."
+        />
+      )
+      : (
+        <>
+          <strong>
+            <FormattedMessage
```

Comment:
The message should be passed as a param to FormattedMessage to allow for languages where `Reason` might not go first.

Is there a styling option aside from `<strong />`?

Reply:
I'll just print the error, similar to how it is done in other places to avoid complications with formatting and translation.